siliconcompiler/tools/vpr/vpr.py

```diff
@@ -26,6 +26,47 @@ def make_docs(chip):
     return chip
 
 
+def assemble_options(chip, tool):
+
+    step = chip.get('arg', 'step')
+    index = chip.get('arg', 'index')
+    # task = chip._get_task(step, index)
+
+    options = []
+
+    topmodule = chip.top()
+    blif = f"inputs/{topmodule}.blif"
+
+    for arch in chip.get('fpga', 'arch'):
+        options.append(arch)
+
+    options.append(blif)
+
+    if 'sdc' in chip.getkeys('input'):
+        options.append(f"--sdc_file {chip.get('input', 'fpga', 'sdc', step=step, index=index)}")
+    else:
+        options.append("--timing_analysis off")
+
+    # Routing graph XML:
+    rr_graphs = chip.get('tool', 'vpr', 'task', 'apr', 'var', 'rr_graph', step=step, index=index)
+    if (len(rr_graphs) == 1):
+        options.append("--read_rr_graph " + rr_graphs[0])
+
+    # ***NOTE: For real FPGA chips you need to specify the routing channel
+    #          width explicitly.  VPR requires an explicit routing channel
+    #          with when --read_rr_graph is used (typically the case for
+    #          real chips).  Otherwise VPR performs a binary search for
+    #          the minimum routing channel width that the circuit fits in.
+    # Given the above, it may be appropriate to couple these variables somehow,
+    # but --route_chan_width CAN be used by itself.
+    num_routing_channels = chip.get('tool', 'vpr', 'task', 'apr', 'var', 'route_chan_width',
+                                    step=step, index=index)
```

Comment:
this gets called on all tasks, please set this correctly for each task. Using `tool, task = chip._get_tool_task(step, index)`